### DISCUSSION TITLE
Using Raven 3.5 ETL to Raven5

### DISCUSSION THREAD

**Discussion by wcappy** [ravendb/ravendb#15756] [General]
Hi i am upgrading a legacy Raven3.5 instance to Raven Cloud (5.4)
I am considering data migration  options due to having a large amount of data that has to be moved, exporting a dump of the Raven3.5 would take took long in our production so i am consider to

**Replicate or ETL to the Raven 5 Cloud instance.**
ETL seems to work locally but with Raven5 requires registration of the certificate is there away of handling this.

I would like to avoid writing a sciript to loadin 3.5 data then store in 5 if possible.

**Comment by arekpalinski** [comment]
Hi,

You have two options:

1) Import from live 3.5 instance:
https://ravendb.net/docs/article-page/4.2/Csharp/migration/server/data-migration#how-to-live-import-data-from-a-running-instance?
https://ravendb.net/docs/article-page/5.4/csharp/studio/database/tasks/import-data/import-from-ravendb

2) One time replication from 3.5 to RavenDB Cloud:
 - in RavenDB 3.5 you can set `Raven/Replication/CertificatePath` and `Raven/Replication/CertificatePassword` and setup the replication to 5.4 instance running in RavenDB Cloud
